server/remote/gitea/helper.go

```diff
@@ -110,15 +111,14 @@ func buildFromPush(hook *pushHook) *model.Build {
 }
 
 func getChangedFilesFromPushHook(hook *pushHook) []string {
-	files := make([]string, 0)
-
+	files := make([]string, 0, len(hook.Commits)*4)
```

Comment:
Why `len(hook.Commits)*4`?

Reply:
to make it faster (less re-alocations), if you guess ca. the size you need it's better.

the idea behind this number is: "a normal number of changed files per commit = 4"

Reply:
PS: if it do not match the exact size it still works as intended

Reply:
```suggestion
  // assume a capacity of 4 changed files per commit
	files := make([]string, 0, len(hook.Commits)*4)
```